identity-aware-proxy/IAPClient/IAPClient.cs

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2018 Google Inc
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// [START generate_iap_request]
+
+using System;
+using System.Collections.Generic;
+using System.IdentityModel.Tokens.Jwt;
+using System.IO;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Security.Claims;
+using Google.Apis.Auth.OAuth2;
+using Google.Apis.Json;
+using Microsoft.IdentityModel.Tokens;
+using Newtonsoft.Json;
+
+namespace GoogleCloudSamples
+{
+    class IAPClient
+    {
+        public static string InvokeRequest(string iapClientId,

```

Comment:
I think that for example code we should follow best practices here, which in this case would mean having async method signature and let the customer of this code to do *.Result or *.Wait if they want to force it to run synchronously.
So perhaps,
public static async string RequestAsync(string iapClientId, string credentialsFilePath, string requestedUri)

Reply:
A surprising number of C# developers do not use or understand async or await.  So we only use them when necessary.  For example, when implementing an MVC Controller method, using async is appropriate.  For a command-line utility, async and await add nothing but complications for the reader.  And readers who are familial with async and await can easily change the code to be async.